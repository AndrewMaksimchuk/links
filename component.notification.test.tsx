import { describe, test, expect } from 'bun:test'
import { renderToReadableStream } from 'hono/jsx/streaming'
import { Notification } from './component.notification'

describe('Component Notification', () => {
    test('status info', async () => {
        const stream = renderToReadableStream(<Notification
            status='info'
            body='Test info status'
            header='Information about some stuff'
            notificationid={0} />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })


    test('status error', async () => {
        const stream = renderToReadableStream(<Notification
            status='error'
            body='Test info status'
            header='Information about some stuff'
            notificationid={0} />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })


    test('status success', async () => {
        const stream = renderToReadableStream(<Notification
            status='success'
            body='Test info status'
            header='Information about some stuff'
            notificationid={0} />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })


    test('status warn', async () => {
        const stream = renderToReadableStream(<Notification
            status='warn'
            body='Test info status'
            header='Information about some stuff'
            notificationid={0} />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })
})
