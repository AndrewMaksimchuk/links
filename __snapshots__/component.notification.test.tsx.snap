// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Notification status info 1`] = `"<div id="notification" hx-swap-oob="afterbegin"><div style="padding:var(--pico-form-element-spacing-vertical) var(--pico-form-element-spacing-horizontal)" data-notificationId="0"><h2 style="border-bottom:solid gray">Information about some stuff</h2><p>Test info status</p><button onclick="this.parentElement.remove()">Close</button></div></div>"`;

exports[`Component Notification status error 1`] = `"<div id="notification" hx-swap-oob="afterbegin"><div style="padding:var(--pico-form-element-spacing-vertical) var(--pico-form-element-spacing-horizontal)" data-notificationId="0"><h2 style="border-bottom:solid red">Information about some stuff</h2><p>Test info status</p><button onclick="this.parentElement.remove()">Close</button></div></div>"`;

exports[`Component Notification status success 1`] = `"<div id="notification" hx-swap-oob="afterbegin"><div style="padding:var(--pico-form-element-spacing-vertical) var(--pico-form-element-spacing-horizontal)" data-notificationId="0"><h2 style="border-bottom:solid green">Information about some stuff</h2><p>Test info status</p><button onclick="this.parentElement.remove()">Close</button></div></div>"`;

exports[`Component Notification status warn 1`] = `"<div id="notification" hx-swap-oob="afterbegin"><div style="padding:var(--pico-form-element-spacing-vertical) var(--pico-form-element-spacing-horizontal)" data-notificationId="0"><h2 style="border-bottom:solid orange">Information about some stuff</h2><p>Test info status</p><button onclick="this.parentElement.remove()">Close</button></div></div>"`;
